### DISCUSSION TITLE
Look into using `anyhow` more intelligently and attaching context

### DISCUSSION THREAD

**Discussion by elsirion** [fedimint/fedimint#3138] [General]
> I think this works, the double error message probably comes from `anyhow` being able to carry [additional context](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#example) but not being used in that way here.

_Originally posted by @elsirion in https://github.com/fedimint/fedimint/issues/1021#issuecomment-1334510856_

**Comment by elsirion** [comment]
We also might want to look into https://crates.io/crates/error-stack

**Comment by cbergqvist** [comment]
Seems like b98a965 went a fair bit of the way to improve error handling for the case related to #1021.

Latest master as of now:
https://github.com/fedimint/fedimint/blob/1ed20ef91e9ec31e3d68602bde8e3f76c25ae252/gateway/ln-gateway/src/lib.rs#L552-L554

**Comment by m1sterc001guy** [comment]
Should we close this or are there additional action items?

**Comment by justinmoon** [comment]
I wonder if this should be a GitHub discussion? There's probably a lot more to say here -- especially about error-stack -- but nothing actionable.
